#!/usr/bin/env python
PACKAGE = "cam_sync"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# enable/disable exposure control
gen.add("enabled", bool_t,   0, "enable/disable auto exposure", False);
gen.add("min_gain", double_t, 0, "auto gain min", 0, 0, 10);
gen.add("max_gain", double_t, 0, "auto gain max", 30, 0, 30);
gen.add("min_shutter", double_t,0,"auto shutter min (msec)", 0.001, 0.0001, 100);
gen.add("max_shutter", double_t, 0, "auto shutter max (msec)", 500, 1, 1000);
gen.add("brightness", double_t, 0, "target brightness (0-255)", 70, 0, 255);
gen.add("brightness_tol", double_t, 0, "target brightness tolerance (0-255)", 10, 0, 255);
gen.add("top_margin", int_t, 0, "top margin (percent)", 0, 0, 100);
gen.add("bottom_margin", int_t, 0, "bottom margin (percent)", 0, 0, 100);
gen.add("wait_frames", int_t, 0, "frames to wait for change", 4, 0, 10);
exit(gen.generate(PACKAGE, "cam_sync", "ExposureControlDyn"))
